### ISSUE TITLE
[java] ImmutableField - false positive with Lombok @Setter

### ISSUE THREAD

**Issue by zman0900** [pmd/pmd#4254]
**Rule:** [ImmutableField](https://pmd.github.io/latest/pmd_rules_java_design.html#immutablefield)

Given a simple class like:

```java
import lombok.Setter;

public class Junk {
    @Setter
    private String value = "abc";

    public String something() {
        return "blah" + value;
    }
}
```

PMD 6.52.0 will report "Private field 'value' could be made final; it is only initialized in the declaration or constructor."  This is a false positive because there is a generated `setValue(String)` method that can modify `value`.
